(docs) update docs/design.md (#717)
cloudposse/geodesic: docs/design.md

* `/etc/motd` is the current "Message of the Day"
* `/localhost` is where we house the local state (like your temporary AWS credentials). This is mounted to your `HOME` directory.
* `/conf` is where all configurations belong. For example, stick your terraform module invocations in this directory. When using `terraform`, the directory names are mapped directly to a bucket prefix for terraform state which includes subdirectories (e.g. `/conf/us-west-2/vpc` will map to a state folder prefix of `us-west-2/vpc`).

## EXTENDING

Geodesic was written to be easily extended. There are a couple of ways to do it. 

You can easily extend the Geodesic shell by creating your repo with a `Dockerfile`. We suggest you have it inherit `FROM geodeisc:latest` (or pin it to a [build number](https://travis-ci.org/cloudposse/geodesic) for stability). If you want to add or modify core functionality, this is the recommended way to do it.

Inside your container, you can replace any of our code with your own to make it behave exactly as you wish. You could even create one dedicated shell per cluster with logic tailored specifically for that cluster.

### Here are some other tips.

> 1. Many of our `Makefiles` do an `-include Makefile.*`, which means, we'll include other `Makefiles` in that directory. To add additional functionality, drop in your `Makefile.something` in that directory.

Reviewer: ```suggestion
1. Many of our `Makefiles` do an `-include Makefile.*`, which means, we'll include other `Makefiles` in that directory. To add additional functionality, simply add a `Makefile.something` file in that directory.
```
